### PR TITLE
More rubocop fixes

### PR DESCRIPTION
**Pull request by alexjfisher**

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -25,12 +19,10 @@ Lint/ParenthesesAsGroupedExpression:
     - 'spec/defines/jenkins_augeas_spec.rb'
     - 'spec/unit/jenkins_plugins_spec.rb'
 
-# Offense count: 3
+# Offense count: 1
 Lint/UselessAssignment:
   Exclude:
-    - 'lib/puppet/provider/jenkins_user/cli.rb'
     - 'spec/helpers/rspechelpers.rb'
```
Comment (alexjfisher) [inline .rubocop_todo.yml]: Sorry, wasn't too sure about this one.

File: spec/unit/puppet/x/spec_jenkins_types.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@
 end # generic namevar
 
 shared_examples 'generic ensurable' do |*allowed|
-  allow ||= [:present, :absent]
+  allowed ||= [:present, :absent]
```
Comment (alexjfisher) [inline spec/unit/puppet/x/spec_jenkins_types.rb]: An actual bug! :)

Comment (bastelfreak) [reply spec/unit/puppet/x/spec_jenkins_types.rb]: bugs \o/

File: lib/puppet/x/jenkins/provider/cli.rb [hunk 3]
```diff
@@ -286,12 +286,8 @@ def self.execute_exceptionify(cmd, options)
     end
 
     begin
-      # return Puppet::Provider.execute(*args)
-      if tmpfile_as_param && options.key?(:stdinfile)
-        return superclass.execute([cmd, options[:stdinfile]].flatten.join(' '), options)
-      else
-        return superclass.execute([cmd].flatten.join(' '), options)
-      end
+      return superclass.execute([cmd, options[:stdinfile]].flatten.join(' '), options) if tmpfile_as_param && options.key?(:stdinfile)
```
Comment (jhoblitt) [inline lib/puppet/x/jenkins/provider/cli.rb]: 👎 I don't like long "run-on" statements.

Comment (alexjfisher) [reply lib/puppet/x/jenkins/provider/cli.rb]: Sorry.  I guess I got a little carried away!  Fixed.

**Pull request merged**